exonum/src/runtime/dispatcher/schema.rs

```diff
@@ -149,17 +149,54 @@ impl Schema<&Fork> {
             return Err(Error::ServiceIdExists);
         }
 
-        let id = spec.id;
-        let name = spec.name.clone();
+        let instance_id = spec.id;
+        let instance_name = spec.name.clone();
+        let pending_status = InstanceStatus::Active;
+
         instances.put(
-            &name,
+            &instance_name,
             InstanceState {
-                spec: spec.clone(),
-                status: InstanceStatus::Pending,
+                spec,
+                status: None,
+                pending_status: Some(pending_status),
             },
         );
-        instance_ids.put(&id, name);
-        self.pending_instances().push(spec);
+        self.pending_instances().put(&instance_name, pending_status);
+        instance_ids.put(&instance_id, instance_name);
+        Ok(())
+    }
+
+    /// Adds information about stopping service instance to the schema.
+    pub(crate) fn initiate_stopping_service(
+        &mut self,
+        instance_id: InstanceId,
+    ) -> Result<(), Error> {
+        let mut instances = self.instances();
+        let mut pending_instances = self.pending_instances();
+
+        let instance_name = self
+            .instance_ids()
+            .get(&instance_id)
+            .ok_or(Error::IncorrectInstanceId)?;
+
+        let mut state = instances
```

Comment:
Micro-nit: If `state` will have methods like `into_active`/`into_stopped` which will return a new object with changed field, we will be able to get rid of the mutable variable here, which IMO will be more `rust`-ish.